### PR TITLE
fix: move autovalue to annotation processor path

### LINKED ISSUE #178: Autovalue should be on the annotation processor classpath
not in dependencies section. Only the annotations belong in the dependencies section. see https://github.com/google/auto/blob/master/value/userguide/index.md

### PR DESCRIPTION
**Pull request by elharo**
@suraj-qlogic @suztomo

fixes #178

### CODE REVIEW

**Comment by elharo** [comment]
2020-07-27T15:15:25.6293721Z [ERROR] com.google.cloud.storage.contrib.nio.CloudStorageFileAttributesTest.testAcl  Time elapsed: 0.451 s  <<< ERROR!
2020-07-27T15:15:25.6302204Z java.nio.file.FileSystemNotFoundException: Provider "gs" not installed
2020-07-27T15:15:25.6310326Z 	at com.google.cloud.storage.contrib.nio.CloudStorageFileAttributesTest.before(CloudStorageFileAttributesTest.java:49)
2020-07-27T15:15:25.6311534Z

**Comment by elharo** [comment]
Just maybe relevant:

 [WARNING] *****************************************************************
[WARNING] * Your build is requesting parallel execution, but project      *
[WARNING] * contains the following plugin(s) that have goals not marked   *
[WARNING] * as @threadSafe to support parallel building.                  *
[WARNING] * While this /may/ work fine, please look for plugin updates    *
[WARNING] * and/or request plugins be made thread-safe.                   *
[WARNING] * If reporting an issue, report it against the plugin in        *
[WARNING] * question, not against maven-core                              *
[WARNING] *****************************************************************
[WARNING] The following plugins are not marked @threadSafe in Storage Parent:
[WARNING] org.codehaus.mojo:clirr-maven-plugin:2.8
[WARNING] Enable debug to see more precisely which goals are not marked @threadSafe.
[WARNING] ***

**Comment by elharo** [comment]
tests are running in parallel and are not using random paths:

```
  @Before
  public void before() {
    CloudStorageFileSystemProvider.setStorageOptions(LocalStorageHelper.getOptions());
    path = Paths.get(URI.create("gs://bucket/randompath"));
    dir = Paths.get(URI.create("gs://bucket/randompath/"));
  }
```

**Comment by frankyn** [comment]
This issue might be related to recent failures. Not quite sure what's going on though as I'm not able to repro this issue locally.

**Comment by frankyn** [comment]
Woot, looks like things are passing now. Pending review from @BenWhitehead when he's back in the office tomorrow.

File: google-cloud-nio/pom.xml
Comment (elharo) [inline google-cloud-nio/pom.xml:115]: 1.7.4 is out now

**Comment by elharo** [comment]
```
com.google.api.client.http.HttpResponseException:
403 Forbidden

POST https://storage.googleapis.com:443/upload/storage/v1/b/bucket/o?uploadType=resumable&name=randompath
{
  "error": {
    "code": 403,
    "message": "it-service-account@gcloud-devel.iam.gserviceaccount.com does not have storage.objects.create access to bucket/randompath.",
    "errors": [
      {
        "message": "it-service-account@gcloud-devel.iam.gserviceaccount.com does not have storage.objects.create access to bucket/randompath.",
        "domain": "global",
        "reason": "forbidden"
      }
    ]
  }
```

**Comment by frankyn** [comment]
Will continue debugging. This looks related to the open failures that exist right now.

**Review by frankyn — APPROVED** [review]

**Comment by frankyn** [comment]
@elharo I found that auto-processor was missing in plugins which helped generate the META-INF/ file that's expected.

If it looks good to you, can we merge this in?

**Review by BenWhitehead — CHANGES_REQUESTED** [review]
Shared config has had auto-value support for some time.

We should be able to remove auto-value and auto-service and use the config from shared config.

See https://github.com/googleapis/java-firestore/pull/221/files for an example of it being used in Firestore.

**Review by elharo — COMMENTED** [review]
@BenWhitehead PTAL. The commit history is becoming convoluted, but I think this is now what you asked for,

**Comment by frankyn** [comment]
Stuck on:
```
[INFO] <<< maven-dependency-plugin:3.1.2:analyze (default-cli) < test-compile @ google-cloud-nio <<<
[INFO] 
[INFO] 
[INFO] --- maven-dependency-plugin:3.1.2:analyze (default-cli) @ google-cloud-nio ---
Warning:  Used undeclared dependencies found:
Warning:     com.google.auto.value:auto-value-annotations:jar:1.7.4:compile
```

Thanks @elharo!

**Comment by frankyn** [comment]
Kokoro integrations failed again. I think it might be a flake and not sure how to reinforce without doing a retry of the tests.

**Comment by frankyn** [comment]
Chatting with @BenWhitehead, the kokoro build change to retry tests will be removed by autosynth...

**Review by BenWhitehead — APPROVED** [review]

**Comment by frankyn** [comment]
We will merge this PR once jobs pass and will follow-up with another solution provided by @BenWhitehead for flaky integration tests to separate tasks. 
I'm still on the hook as part of On Duty.

**Pull request merged**
